packages/language-server/src/completion/completionUtils.ts

```diff
@@ -406,11 +407,23 @@ export function filterSuggestionsForBlock(
     // Ignore commented lines
     if (!item.startsWith('//')) {
       // TODO we should also remove the other suggestions if used (default()...)
-      if (item.includes('@id')) {
-        suggestions = suggestions.filter((attribute) => !attribute.label.includes('id'))
-      }
+      // * Filter already-present attributes that can't be duplicated
+      ;['@id', '@@map', '@@ignore', '@@schema'].forEach((label) => {
```

Comment:
Should this also be `@@id` for consistency (and then filter `@id` separately below like `@ignore`)? Or am I missing something and it will not be equivalent?

Reply:
I wasn't sure whether to change that or not. That intuitively makes more sense as it's technically for tracking _block_ attributes but `@id` is also in a way a ""block attribute""
<img width="815" alt="image" src="https://github.com/prisma/language-tools/assets/29753584/737eac10-f015-4153-96e6-ea177bc8e91c">

Reply:
I carefully read your explanation 5 times but I fail to reason how the current logic is any different than adding `@ignore` and `@@id` to the list of attributes to de-duplicate.

One more thing I'm wondering is why we're not generalizing the logic to filter down _any_ suggestions that are already present on the list:

```ts
suggestions = suggestions.filter(suggestion => !item.includes(label))
```

Even if that's not an option because of some edge cases, the block could be refactored as:

```ts
const attributesToDedup = ['@id', '@@map', '@@ignore', '@@schema'];

suggestions = suggestions.filter(suggestion => item.includes(suggestion.label) && !attributesToDedup.includes(suggestion.label));
```

which would reduce the 3*N filters worst-case scenario down to a single one.

Lastly, I never touched this piece of code so it's likely I'm missing something here!

Reply:
Note: I will merge this as is because it's a good improvement already, and Soph can take a look at this when they are back next week.